src/state/opamUpdate.ml

```diff
@@ -510,7 +510,8 @@ let download_package_source st nv dirname =
     | None   -> Done None
     | Some u ->
       (OpamRepository.pull_tree (OpamPackage.to_string nv)
-        ~cache_dir ~cache_urls ?subpath:(OpamFile.URL.subpath u)
+        ~cache_dir ~cache_urls ?working_dir
+        ?subpath:(OpamFile.URL.subpath u)
```

Comment:
tbh, i don't remember why I wrote this propagation of working dir a2e51d3f, but at that moment I'm pretty sure that I had a very good reason :D
This function `download_package_source` is not called on the update operation itself, but on graph action fetch operation (and `opam source`).

Reply:
Ah, right, the argument is passed to the `download_package_source` function, which is specific to the package, so I guess it's just a matter for the caller to specify it correctly!

The reason why this wasn't needed is that working_dir is assumed to be only on dev packages, and these are updated through different functions (`fetch_dev_package`, etc.), and then skipped by `download_package`. This could have been incorrect if there are cases where we reach the Fetch action processing and the dev package wasn't yet cached, did you hit such a case ?

Reply:
I don't find a good reason to keep it as i don't remember why I added it, and don't find one now. I ran some tests, and everything is ok without. Reverted.